boltdesignsystem/bolt
# Core | JS | Move 'optimizedResize' into Core

## Jira

http://vjira2:8080/browse/BDS-1882

## Summary

Move `optimizedResize` event into Bolt Core and rename it `throttledResize`. Add `debouncedResize` event.

## Details
`optimizedResize` was duplicated in a several components. I've refactored into a single file in Core. I've also included a new `debouncedResize` Custom Event, which is currently only used by image but can be used by any other component.

### Fixes/Improvements
- `throttledResize` is now only included once on the page, even if multiple components use it.
- `debouncedResize` event listeners are now properly removed when Images are disconnected. Previously, `this.onResize` was wrapped in `debounce()` and not removed on `disconnecting()` ([see code](https://github.com/boltdesignsystem/bolt/pull/1531/files#diff-9d632ea9e7c654cc2b08adafc542ecf3L99-L149)).

## How to test

- Review files changed
- Checkout `feature/move-optimized-resize-to-core` locally
- Test the following components cross-browser, especially what happens on resize, and verify there are no regressions:
  - Carousel
  - Image
  - Nav Priority
  - Tabs
